### PR TITLE
fix #369 reset SHA-1 challenge method

### LINKED ISSUE #369: Reconnection on the same Strophe instance causes a null reference exception
Given an XMPP server uses SCRAM-SHA1 authentication scheme, a subsequent reconnection performed on the same Strophe.Connection instance causes an 'cannot call match method of null' exception at the following line:

https://github.com/strophe/strophejs/blob/master/src/core.js#L3380

As far as I can understand the root cause of the issue, the SASLSHA1 authentication mechanism is self-modifying in terms of overwriting its `onChallenge` method implementation upon the first call to the method's `onChallenge`.

As long as this overwrite is not rolled back on invoking `connection.reset()`, any subsequent authentication attempts that assume a client-first challenge would result in the aforementioned error.

Would you please clarify if this is intentional, and one is not supposed to reuse a Strophe.Connection instance for re-connecting to the server? And, if this is a breaking change introduced in recent versions of Strophe, because I believe this issue did not exist in Strophe v1.2.x ?

### PR DESCRIPTION
**Pull request by mightymop**
fix #369 
resets SHA-1 challenge method chain in reset method so that a new connection will be working...

### CODE REVIEW

**Comment by mightymop** [comment]
review plz :)

File: src/core.js [hunk 1]
```diff
@@ -1602,6 +1602,7 @@ Strophe.Connection = class Connection {
         // SASL
         this.do_session = false;
         this.do_bind = false;
+        this.mechanisms["SCRAM-SHA-1"].reset();
```
Comment (jcbrand) [inline src/core.js:1605]: There's no guarantee that this mechanism exists at runtime.

**Comment by jcbrand** [comment]
I believe the issue this MR was made to fix is now fixed here: https://github.com/strophe/strophejs/commit/ae0131a48a2080c51a8f6f312ecc46361e96b098

**Comment by mightymop** [comment]
i closed it^^

**Pull request closed without merging**